api/src/main/java/org/openmrs/module/kenyaemr/reporting/builder/mchms/ANCRegisterReportBuilder.java

```diff
@@ -105,6 +105,7 @@ protected DataSetDefinition datasetColumns() {
         dsd.addColumn("Number of ANC Visits", new ANCNumberOfVisitsDataDefinition(),"");
         dsd.addColumn("Name", nameDef, "");
         dsd.addColumn("Telephone No", new PersonAttributeDataDefinition(phoneNumber), "");
+        dsd.addColumn("Village", new ANCVillageDataDefinition(), null);
```

Comment:
You may need to consider using "**PersonAddressCalculation**". However, your approach is still okay but more work.